### PR TITLE
Feat(CLI): delete user

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
This PR: 
- adds functionality to delete TB users via the CLI
- adds functionality to list users via the CLI

### How to prepare for test
- [x] ssh to hasta (depending on type of change)
- [x] activate stage: `us`
- [x] request trailblazer-stage on hasta: `paxa`
- [x] install on stage:
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-trailblazer-stage.sh [THIS-BRANCH-NAME]`
- [x] ssh to clinical-db (depending on type of change)
- [x] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh [THIS-BRANCH-NAME]`

### How to test archiving users
- [x] login to hasta
- [x] do su
- [x] archive user `trailblazer archive-user [email]`

### Expected test outcome
- [x] check that the user can't use trailblazer functionality when logged into cigrid-stage

### How to test listing users
- [x] login to hasta
- [x] do su
- [x] trailblazer users --name rik --email life

### Expected test outcome
- [x] check that the users listed have 'rik' in their names and 'life' in their emails 
- [x] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] code approved by
- [x] tests executed by PG
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: trailblazer/store/api.py [hunk 1]
```diff
@@ -192,6 +192,11 @@ def add_user(self, name: str, email: str) -> models.User:
         self.add_commit(new_user)
         return new_user
 
+    def add_user(self, user: models.User) -> None:
```
Comment (Mropat) [inline trailblazer/store/api.py:195]:
```suggestion
    def delete_user(self, user: models.User) -> None:
```

**Comment by patrikgrenfeldt** [comment]
Each analysis has a link to the user that initiated it:
image

Either:
- Remove restriction (possibly adding a col for text name of initiator) 
- Archive users and make it impossible to login for archived users

**Comment by patrikgrenfeldt** [comment]
Test result listing users:
```
trailblazer users --name rik --email life
Listing users in database:
{'name': 'XXXrik XXXX', 'created_at': datetime.datetime(XXXX), 'email': 'XXXX.XXXX@scilifelab.se', 'id': 1, 'is_archived': False, 'google_id': None, 'avatar': None}
{'name': 'XXXXrik XXXX', 'created_at': datetime.datetime(XXXX), 'email': 'XXXX.XXXX@scilifelab.se', 'id': 2, 'is_archived': True, 'google_id': None, 'avatar': None}
{'name': 'XXXXrik XXXX', 'created_at': datetime.datetime(XXXX), 'email': 'XXXX.XXXX@scilifelab.se', 'id': 3, 'is_archived': False, 'google_id': None, 'avatar': None}
```
🆗 users filtered and listed

Test result archiving user:
```
 trailblazer archive-user 'XXXX.XXXX@scilifelab.se'
User archived: XXXX.XXXX@scilifelab.se
```
🆗 user archived

Test result trailblazer usage:
image
🆗 trailblazer functionality not accessible in cigrid

**Comment by Mropat** [comment]
Great, but we also need the archival to do something. So I think we need to add logic into the authentication part of the API to deny access to archived users. 
Also the cli command archive user will archive, but we dont have a way to reverse archival. So maybe it can be reworked into "user archive --enable/--disable" to cover all bases

**Comment by patrikgrenfeldt** [comment]
Test after changes: 
image
🆗 still not possible to use TB funcionality in Cigrid

**Comment by patrikgrenfeldt** [comment]
Migrated database:
image

Deployed to clinical-db:
image

Deployed to hasta:
image

**Pull request merged**